images/echoserver/echoserver.go

```diff
@@ -181,6 +181,18 @@ func writeEchoResponseHeaders(w http.ResponseWriter, headers http.Header) {
 
 }
 
+func logRequestAssertions(rs RequestAssertions, w http.ResponseWriter, headers http.Header) {
+	js, err := json.Marshal(rs)
+	if err != nil {
+		processError(w, err, http.StatusInternalServerError)
+		return
+	}
+	_, ok := headers["X-Echo-Log-Request"]
+	if ok {
+		fmt.Printf("The request is: %v\n", string(js))
+	}
```

Comment:
It would seem we don't need to spend cycles marshaling the JSON for requests which don't enable the logging via the headers?

```suggestion
	_, ok := headers["X-Echo-Log-Request"]
	if ok {
		js, err := json.Marshal(rs)
		if err != nil {
			processError(w, err, http.StatusInternalServerError)
			return
		}

		fmt.Printf("The request is: %v\n", string(js))
	}
```

Reply:
Thank you for the suggestion, the change accepted